src/tokenizer/types.js

```diff
@@ -80,7 +80,7 @@ export const types = {
   doubleColon: new TokenType("::", { beforeExpr }),
   dot: new TokenType("."),
   question: new TokenType("?", { beforeExpr }),
-  arrow: new TokenType("=>", { beforeExpr }),
+  arrow: new TokenType("=>", { beforeExpr, startsExpr }),
```

Comment:
Yes, unfortunately this is all declarative code, running outside the context of a particular `this` or parser configuration. Would require major changes to the way babylon initializes itself.